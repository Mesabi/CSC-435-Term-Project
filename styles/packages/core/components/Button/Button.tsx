import styled, { th } from '@xstyled/styled-components';

const Button = styled.button`
  background-color: material;
  padding: 7 20 5;
  border: none;
  color: materialText;

  font-size: 12px;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
    inset 0 0 0 1px ${th('colors.borderDark')},
    1px 1px 0 0px ${th('colors.borderDarkest')};

  &:disabled {
    color: materialTextDisabled;
  }

  &:focus {
    outline: ${th('space.1')}px dotted ${th('colors.materialText')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
      inset -0.5px -0.5px 0px 1px ${th('colors.borderDark')},
      1px 1px 0 1px ${th('colors.borderDarkest')};
  }

  &:active {
    padding: 8 20 4;

    outline: ${th('space.1')}px dotted ${th('colors.borderDarkest')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 0 0 0 1px ${th('colors.borderDark')},
      0 0 0 1px ${th('colors.borderDarkest')};
  }
`;

Button.defaultProps = {
  children: 'Ok',
};

export default Button;
