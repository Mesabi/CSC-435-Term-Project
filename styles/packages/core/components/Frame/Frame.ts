import styled, { system, SystemProps } from '@xstyled/styled-components';

export interface FrameProps extends SystemProps {}

const Frame = styled.div<FrameProps>(system);

Frame.defaultProps = {
  w: 'auto',
  h: 'auto',
  bg: 'material',
  boxShadow: 'out',
};

export default Frame;
