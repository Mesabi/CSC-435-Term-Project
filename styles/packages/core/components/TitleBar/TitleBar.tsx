import React from 'react';
import styled, { th } from '@xstyled/styled-components';

import Button from '../Button';
import Frame, { FrameProps } from '../Frame/Frame';

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Button).attrs({
  as: 'li',
})`
  display: flex;
  align-items: center;
  justify-content: center;

  margin-left: 2;
  padding: 0;

  width: 17px;
  height: 14px;

  min-width: 0;
  font-weight: 600;
  font-size: 10;

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:first-child {
    margin-right: 0;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
      inset -1px -1px 0px 1px ${th('colors.borderDark')};
  }
`;

Option.displayName = 'Option';

interface TitleBarBackgroundProps
  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'color'>,
    FrameProps {
  active?: boolean;
  icon?: React.ReactElement;
  title?: string;
}

const TitleBarBackground = styled(Frame)<{ active?: boolean }>`
  height: 18;
  margin-bottom: 2;

  padding: 2;

  display: flex;

  box-shadow: none;

  img {
    width: 15;
    height: 14;
    margin-right: 4;
  }

  background: ${({ active, theme }) =>
    active
      ? theme.colors.headerBackground
      : theme.colors.headerNotActiveBackground};
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
`;

interface TitleBarOptions {
  Option: typeof Option;
  OptionsBox: typeof OptionsBox;
}

export interface ITitleBarProps
  extends TitleBarBackgroundProps,
    TitleBarOptions {}

const TitleBarRenderer: React.ForwardRefRenderFunction<
  HTMLDivElement,
  TitleBarBackgroundProps
> = (
  { children, title = 'UNKNOWN.EXE', icon, active = true, className },
  ref: React.Ref<HTMLDivElement>,
) => (
  <TitleBarBackground
    active={active}
    color={active ? 'headerText' : 'headerNotActiveText'}
    ref={ref}
  >
    {icon && icon}
    <Title className={className}>{title}</Title>

    {children}
  </TitleBarBackground>
);

const TitleBar = Object.assign(
  React.forwardRef<HTMLDivElement, TitleBarBackgroundProps>(TitleBarRenderer),
  {
    Option,
    OptionsBox,
  },
) as ITitleBarProps & typeof TitleBarRenderer;

export default TitleBar;
