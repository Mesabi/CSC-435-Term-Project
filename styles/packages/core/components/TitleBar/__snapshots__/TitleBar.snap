// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar /> Snapshot should match snapshot 1`] = `
.c0 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c3 {
  color: #c6c6c6;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c1 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c4 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #7f787f;
}

.c4 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

<div>
  <div
    class="c0 c1"
    color="headerText"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c2"
    >
      test.exe
    </div>
  </div>
  <div
    class="c3 c4"
    color="headerNotActiveText"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c2"
    >
      test.exe
    </div>
  </div>
  <div
    class="c0 c1"
    color="headerText"
  >
    <div
      class="c2"
    >
      UNKNOWN.EXE
    </div>
  </div>
  <div
    class="c3 c4"
    color="headerNotActiveText"
  >
    <div
      class="c2"
    >
      UNKNOWN.EXE
    </div>
  </div>
</div>
`;

exports[`<TitleBar /> Snapshot should match snapshot with all Options 1`] = `
.c4 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c4:disabled {
  color: #868a8e;
}

.c4:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c4:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:first-child {
  margin-right: 0;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c1 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

<div>
  <div
    class="c0 c1"
    color="headerText"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c2"
    >
      test.exe
    </div>
    <ul
      class="c3"
    >
      <li
        class="c4 c5"
      >
        ?
      </li>
      <li
        class="c4 c5"
      >
        x
      </li>
    </ul>
  </div>
</div>
`;
